report_thesis/src/sections/methodology/testing_validation.tex

```diff
@@ -0,0 +1,223 @@
+\subsection{Validation and Testing Procedures for Model Evaluation}\label{subsec:validation_testing_procedures}
+This section describes the validation and testing procedures that each of our experiments adhere to.
+Selecting appropriate testing procedures is crucial for ensuring the validity and reliability of the results.
+For that reason, we delineate a methodological approach that ensures our models are accurate and generalizable.
+
+We have chosen to test and evaluate all our experiments using both cross-validation and a separate test set.
+Evaluating results solely on the test set could lead to models that are overly specialized to the test set.
+This occurs when one searches for the optimal configuration of hyperparameters specifically tailored to the test set.
+However, our objective is to develop models that demonstrate high accuracy and robustness, even on entirely unseen data.
+To achieve this, we employ k-fold cross-validation to ensure our models have high generalizability, thereby increasing the likelihood that they will perform as expected on new data.
+
+We use an $80\%/20\%$ split for training and testing sets. The training set is further subdivided into $k$ folds, which are used for cross validation.
+
+While we employ conventional techniques like holdout sets and k-fold cross validation, \gls{libs} imposes additional challenges to the process.
+
+One of the primary challenges is preventing data leakage.
+As per concentration matrix $\mathbf{C}$ in Section~\ref{sec:problem_definition}, each target only has one ground truth concentration value per oxide.
+However, each target is shot at multiple locations, resulting in multiple instances of the same target in the dataset, as shown in Table~\ref{tab:final_dataset_example}.
+Although the intensity values vary for each location, they fundamentally represent different measurements of the same target.
+If we were to randomly split the dataset, some locations from a target could end up in the testing set while others remain in the training set.
+This would cause data leakage, as the testing set would no longer consist solely of unseen targets.
+To prevent this, we ensure that each target is represented only once in the dataset by grouping data from all locations on a given target.
+
+Furthermore, the limited availability we mention in Section~\ref{sec:problem_definition} of data poses another significant challenge.
+The dataset we use consists of 408 samples. This is a large dataset by \gls{libs} standards.
+As mentioned, the collection process is difficult, making datasets relatively small in general.
+Additionally, there are only a few samples with extreme concentration values for the oxides in the targets.
+These extreme values act as outliers, presenting concentrations significantly different from the rest of the data points.
```

Comment:
> In statistics, an outlier is a data point that differs significantly from other observations. An outlier may be due to a variability in the measurement, an indication of novel data, or it may be the result of experimental error; the latter are sometimes excluded from the data set.

```suggestion
These extreme values act as benign outliers, presenting concentrations significantly different from the rest of the data points.
```

Reply:
Resolved